evm/src/cpu/contextops.rs

```diff
@@ -326,18 +234,26 @@ fn eval_ext_circuit_set<F: RichField + Extendable<D>, const D: usize>(
         let constr = builder.mul_extension(lv.op.context_op, diff);
         yield_constr.constraint(builder, constr);
     }
-    // The new top is loaded in memory channel 3, if the stack isn't empty (see eval_packed).
+    // The new top is loaded in memory channel 2, if the stack isn't empty (see eval_packed).
     for (&limb_new_top, &limb_read_top) in new_top_channel
         .value
         .iter()
-        .zip(lv.mem_channels[3].value.iter())
+        .zip(lv.mem_channels[2].value.iter())
     {
         let diff = builder.sub_extension(limb_new_top, limb_read_top);
         let prod = builder.mul_extension(lv.general.stack().stack_inv_aux_2, diff);
         let constr = builder.mul_extension(lv.op.context_op, prod);
         yield_constr.constraint(builder, constr);
     }
 
+    // Unused channels.
+    for i in 1..NUM_GP_CHANNELS {
+        if i != 2 {
+            let channel = lv.mem_channels[i];
+            let constr = builder.mul_extension(filter, channel.used);
+            yield_constr.constraint(builder, constr);
+        }
+    }
```

Comment:
This is something we do a fair bit (repeated code). Perhaps a candidate for moving to its own function? And it looks like it could be expensive too, should it use a parallel iterator (can it?)?

Reply:
I agree with code rep, it could be wrapped in a utility function. As for parallel iterator, I think the rayon overhead would be greater than the gains here.

Reply:
I had a stab at this with:

```rust
fn unused_channels<P: PackedField>(
    mem_channels: &[MemoryChannelView<P>],
    yield_constr: &mut ConstraintConsumer<P>,
    filter: P,
) {
    mem_channels
        .iter()
        .skip(2)
        .for_each(|channel| yield_constr.constraint(filter * channel.used))
}
```
…but there's something wrong so leaving it to the experts. :/

Reply:
It should be `skip(3)`, we're starting at 1 and skipping when `i == 2`, effectively ignoring the first 3 elements. Otherwise looks good to me (actually I think it's even more correct, as the code above does not constrain the last `mem_channel` of `lv`, while yours does.